### PR TITLE
Add error handling

### PR DESCRIPTION
**Pull request by garuppel**

### CODE REVIEW

**Comment by garuppel** [comment]
> I think I need to revise the code again. Currently, the network interfaces and load balancers are not discovered if the virtual machine discovery fails.

DONE

File: service/discovery/azure/compute.go [hunk 1]
```diff
@@ -70,10 +70,31 @@ func (d *azureComputeDiscovery) List() (list []voc.IsResource, err error) {
 	}
 
 	// Discover virtual machines
+	virtualMachines, _ := d.discoverVirtualMachines()
```
Comment (oxisto) [inline service/discovery/azure/compute.go:73]: Hm. This basically ignores the error in a higher level than before :)

Should be split this into individual discoverers?

File: service/discovery/azure/compute.go [hunk 2]
```diff
@@ -99,12 +129,20 @@ func (d *azureComputeDiscovery) List() (list []voc.IsResource, err error) {
 		list = append(list, s)
 	}
 
-	// Discover Load Balancer
-	// TODO Client to get load balancer
+	return list, err
+}
+
+// Discover Load Balancer
+func (d *azureComputeDiscovery) discoverLoadBalancer() ([]voc.IsResource, error) {
```
Comment (oxisto) [inline service/discovery/azure/compute.go:136]: I have moved load balancers out of the `Compute` resource, so a separate discoverer would be more appropriate.

**Review by oxisto — CHANGES_REQUESTED** [review]

File: service/discovery/azure/compute.go (hunk 1 above)
Comment (garuppel) [reply service/discovery/azure/compute.go:73]: Why? The error was already handled in the discoverVirtualMachines() function.

**Review by oxisto — APPROVED** [review]
LGTM.

Would be nice if you could not modify the pb.go files. Apart from that looks good.

**Pull request merged**